bin/cron/delete_accounts

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/db'
+require 'cdo/only_one'
+
+OPEN_ENDED_LEVEL_TYPES = %w(
+  Applab
+  FreeResponse
+  Gamelab
+  Weblab
+).freeze
+
+# Deletes all projects and project data associated with a user.
+# @param [Integer] user_id for which to delete projects and project data.
+def delete_projects_and_project_data(user_id)
+  # TODO(asher): Delete channel-backed projects.
+
+  GalleryActivity.where(user_id: user_id).delete_all
+end
+
+# @param [Integer] user_id for which to delete free response answers.
+def delete_free_response_answers(user_id)
+  UserLevels.where(user_id: user_id).find_each do |user_level|
+    if OPEN_ENDED_LEVEL_TYPES.include? user_level.level.type
+      user_level.level_destroy
+    end
+  end
+end
+
+# @param [Integer] user_id for which to delete orphaned students.
+def delete_orphaned_students(user_id)
+  section_ids = Section.with_deleted.
+    where(user_id: user.id).
+    where(login_type: [Section::LOGIN_TYPE_PICTURE, Section::LOGIN_TYPE_WORD]).
+    pluck(:id)
+  Follower.with_deleted.where(section_id: section_ids).find_each do |follower|
+    next unless student_user = follower.student_user
+    next unless student_user.provider == User::PROVIDER_SPONSORED
+    next unless Follower.where(student_user: student_user).count == 1
+
+    # TODO(asher): Determine whether this should be delete_user_and_dependencies(student_user).
+    student_user.destroy!
+  end
+end
+
+# Anonymizes the user by deleting various pieces of PII and PPII from the User and UserGeo models.
+# @param [User] user to be anonymized.
+def anonymize_user(user)
+  # Note that we explicitly want to keep users.user_type, users.gender, users.locale,
+  # users.birthday, users.total_lines, users.sign_in_count, users.last_sign_in_at,
+  # users.current_sign_in_at.
+
+  # Delete various pieces of information stored in the JSON `properties` blob.
+  user.ops_first_name = nil if user.ops_first_name
+  user.ops_last_name = nil if user.ops_last_name
+  user.district_id = nil if user.district_id
+  user.ops_school = nil if user.ops_school
+  user.last_seen_school_interstitial = nil if user.user.last_seen_school_interstitial
+  user.save!
+
+  user.reload.update!(
+    name: '',
+    username: '',
```

Comment:
For better or for worse, there are unique indexes on `username`, `email`, etc. How do we want to manipulate these fields while maintaining the DB constraints? Do we want to relax the DB constraints?